htdocs/core/class/commonobject.class.php

```diff
@@ -1170,6 +1170,11 @@ public function delete_contact($rowid, $notrigger = 0)
 		// phpcs:enable
 		global $user;
 
+		$result = $this->call_trigger(strtoupper($this->element).'_BEFORE_DELETE_CONTACT', $user);
```

Comment:
A definition of a trigger is that action is inside the transaction so after the begin.
There is no need for _BEFORE_ trigger for a delete because for a delete action trigger must always be executed before the deletion of main object.

Creation:
- Create
- Trigger create

Deletion = this opossite order so:
- Trigger delete
- Delete

Reply:
hum ok but please have a look to my example, how could i get the id of the contact who is deleted ?

yes main object (fichinter) is not yet deleted but what about the contact ? who could i get the deleted contact to send him a notification, mail or whatelse about the fact he is no more linked to that intervention ?